research/format.markdown

```diff
@@ -0,0 +1,138 @@
+## Draft 2 to Draft 3:
+
+> Behaviour: Not required to validate → MAY validate(but aren’t required to)
+
+**\-**  **street-address** : This should be a street address.
+
+**\-**  **locality** : This should be a city or town.
+
+**\-**  **region** : This should be a region (a state in the US, province in Canada, etc.)
+
+**\-**      **postal-code** : This should be a postal code (AKA zip code).
+
+**\-**      **country** : This should be the name of a country.
+
+
+**\+**     **host-name** : This SHOULD be a host-name.  
+
+
+
+## Draft 3 to Draft 4:
+
+> MAY validate → 
+      a. they SHOULD implement validation for attributes defined below;
+      b. they SHOULD offer an option to disable validation for this keyword.
+      c. Implementations MAY add custom format attributes but not peer implementations.
+
+**\-**     **date** : This SHOULD be a date in the format of YYYY-MM-DD.
+
+**\-**     **time** : This SHOULD be a time in the format of hh:mm:ss.
+
+**\-**     **utc-millisec** : This SHOULD be the difference, measured in milliseconds, between the specified time and midnight, 00:00 of January 1, 1970 UTC. The value SHOULD be a number (integer or float).
+
+**\-**     **regex** : A regular expression, following the regular expression specification from ECMA 262/Perl 5.
+
+**\-**     **color** : This is a CSS color (like "#FF0000" or "red"), based on CSS 2.1 (Hickson, I., Lie, H., Çelik, T., and B. Bos, “Cascading Style Sheets Level 2 Revision 1 (CSS 2.1) Specification,” July 2007.) [W3C.CR‑CSS21‑20070719].
+
+**\-**     **style** : This is a CSS style definition (like "color: red; backgroundcolor:#FFF"), based on CSS 2.1 (Hickson, I., Lie, H., Çelik, T., and B. Bos, “Cascading Style Sheets Level 2 Revision 1 (CSS 2.1) Specification,” July 2007.) [W3C.CR‑CSS21‑20070719].
+
+**\-**     **phone** : This SHOULD be a phone number (format MAY follow E.123).
+
+### Changes: 
```

Comment:
Please no `:` in titles. A title already introduction a section by definition

Reply:
Also, having a "Changes" subsection everywhere is a good idea to clarify the before/after. We should have it everywhere for consistency

Reply:
Yeah, I like the style of these tables. I would extend it a bit like this:

- First column: format name
- Second column: spec description before (quoting + link)
- Third column: spec description after (quoting + link) (or REMOVED?)
- Fourth column: our own notes and concise explanation of the change